lib/danger/danger_core/plugins/dangerfile_bitbucket_server_plugin.rb

```diff
@@ -0,0 +1,179 @@
+      "bitbucket_server"
+    end
+
+    def initialize(dangerfile)
+      super(dangerfile)
+      @bs = dangerfile.env.request_source
+    end
+
+    # @!group Bitbucket Server Misc
+    # The hash that represents the PR's JSON. For an example of what this looks like
+    # see the [Danger Fixture'd one](https://raw.githubusercontent.com/danger/danger/master/spec/fixtures/bitbucket_server_api/pr_response.json).
+    # @return [Hash]
+    def pr_json
+      @bs.pr_json
+    end
+
+    # @!group PR Metadata
+    # The title of the Pull Request.
+    # @return [String]
+    #
+    def pr_title
+      @bs.pr_json[:title].to_s
+    end
+
+    # @!group PR Metadata
+    # The body text of the Pull Request.
+    # @return [String]
+    #
+    def pr_description
+      @bs.pr_json[:description].to_s
+    end
+    alias pr_body pr_description
+
+    # @!group PR Metadata
+    # The username of the author of the Pull Request.
+    # @return [String]
+    #
+    def pr_author
+      @bs.pr_json[:author][:user][:slug].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged into.
+    # @return [String]
+    #
+    def branch_for_base
+      @bs.pr_json[:toRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # A href that represents the current PR
+    # @return [String]
+    #
+    def pr_link
+      @bs.pr_json[:links][:self].flat_map { |l| l[:href] }.first.to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged from.
+    # @return [String]
+    #
+    def branch_for_head
+      @bs.pr_json[:fromRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The base commit to which the PR is going to be merged as a parent.
+    # @return [String]
+    #
+    def base_commit
+      @bs.pr_json[:toRef][:latestCommit].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The head commit to which the PR is requesting to be merged from.
+    # @return [String]
+    #
+    def head_commit
+      @bs.pr_json[:fromRef][:latestCommit].to_s
+    end
+
+    # @!group Bitbucket Server Misc
+    # Returns a list of HTML anchors for a file, or files in the head repository.
+    # It returns a string of multiple anchors if passed an array.
+    # @param    [String or Array<String>] paths
+    #           A list of strings to convert to github anchors
+    # @param    [Bool] full_path
+    #           Shows the full path as the link's text, defaults to `true`.
+    #
+    # @return [String]
+    #
+    def html_link(paths, full_path: true)
+      paths = [paths] unless paths.kind_of?(Array)
+      commit = head_commit
+      repo = pr_json[:fromRef][:repository][:project][:links][:self].flat_map { |l| l[:href] }.first
+
+      paths = paths.map do |path|
+        url_path = path.start_with?("/") ? path : "/#{path}"
+        text = full_path ? path : File.basename(path)
+        create_link("#{repo}/blob/#{commit}#{url_path}", text)
```

Comment:
I checked if with our bitbucket server installation. Its for example:
`https://stash.example.com/projects/IOS/repos/FancyApp/browse/Classes/Main%20Categories/Feed/FeedViewController.m?at=bd87fd739d67a6d64ef4d2f5a41b61449ba21bf8#200`

* `https://stash.example.com/projects/IOS/repos/FancyApp/browse` is `pr_json[:fromRef][:repository][:links][:self].flat_map { |l| l[:href] }.first`
* `Classes/Main%20Categories/Feed/FeedViewController.m` is `path` with escaped spaces
* the `at` parameter can be a commit or branch name
* `#200` is the line number